src/views/Dashboard/components/AddAnnouncement/index.view.tsx

```diff
@@ -1,49 +1,66 @@
-import React, { useState } from "react"
+import React, { useRef, useState } from "react"
 import { useAuth0 } from "@auth0/auth0-react"
 import { postAnnouncement } from "utils/Api"
 import CoolDownButton from "components/Button/CoolDownButton"
 import "./index.scss"
 
 const AddAnnouncement: React.FC = () => {
   const { getAccessTokenSilently } = useAuth0()
-  const [message, setMessage] = useState<string>("")
-  const [title, setTitle] = useState<string>("")
+
   const [note, setNote] = useState<string>("")
   const [disabled, setDisabled] = useState<boolean>(false)
   const [display, setDisplay] = useState<boolean>(false)
 
+  // references
+  const titleRef = useRef<HTMLInputElement>(null)
+  const messageRef = useRef<HTMLTextAreaElement>(null)
+
   const onSubmit = async () => {
     // event.preventDefault()
+    if (!titleRef.current || !messageRef.current) {
+      setNote("We lost your input box...")
+      return
+    }
+
+    const title = titleRef.current.value
+    const message = messageRef.current.value
+
     setDisabled(true)
     if (title.length <= 0 || title.length >= 50) {
-      setNote("Invalid Title")
+      setNote("Invalid title")
       setDisabled(false)
       return
     }
+
     if (message.length <= 0 || message.length >= 200) {
-      setNote("Invalid Message")
+      setNote("Invalid message")
       setDisabled(false)
       return
     }
+
+    // TODO: we need punctuations and line breaks right?
     if (!/^[a-zA-Z0-9 ]+$/.test(title) || !/^[a-zA-Z0-9 ]+$/.test(message)) {
-      setNote("Alphanumeric Characters Only")
+      setNote("Alphanumeric characters only")
       setDisabled(false)
       return
     }
```

Comment:
We can update the regex for this now or do this separately later

Reply:
Let's update this later, since I don't know the regex that would match ascii characters off the top of my head.